### PR TITLE
Fix `take_along_axis` with TensorFlow with non-int32 indices.

### PR DESCRIPTION
**Pull request by hertschuh**
Change https://github.com/keras-team/keras/pull/21239 broke one use case when the axis dimension is dynamic, the type of the indices is not int32, and the op is run in graph mode.

Note that the additional unit tests don't actually cover this.

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
Thanks for the fix!
